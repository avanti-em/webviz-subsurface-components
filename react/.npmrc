engine-strict=true
ignore-scripts=true
CYPRESS_INSTALL_BINARY=0