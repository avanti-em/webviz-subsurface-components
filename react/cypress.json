{
    "baseUrl": "http://localhost:6006/?path=/story/deckglmap--default",
    "video": true,
    "screenshotsFolder": "./cypress/snapshots/actual",
    "trashAssetsBeforeRuns": true,
    "env": {
        "failSilently": false,
        "ALWAYS_GENERATE_DIFF": false,
        "type": "base"
    }
}
