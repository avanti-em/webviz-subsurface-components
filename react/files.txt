cypress-visual-screenshots/comparison/defaultMap.cy-default-map-story.png
cypress-visual-screenshots/diff/defaultMap.cy-default-map-story.png
files.txt
