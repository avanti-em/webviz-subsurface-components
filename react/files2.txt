cypress-visual-screenshots/diff/defaultMap.cy-default-map-story.png
