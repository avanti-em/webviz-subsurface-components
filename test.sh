if [[ -d react/cypress/snapshots/diff ]]
then
    echo "::set-output name=docs_changed::true"
else
    echo "::set-output name=docs_changed::false"
fi