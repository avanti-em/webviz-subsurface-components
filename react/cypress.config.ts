import { defineConfig } from "cypress";
const getCompareSnapshotsPlugin = require("cypress-image-diff-js/dist/plugin");

export default defineConfig({
  component: {
    devServer: {
      framework: "create-react-app",
      bundler: "webpack",
    },
    video:false,
    setupNodeEvents(on, config) {
      getCompareSnapshotsPlugin(on, config);
    }
  },
});
