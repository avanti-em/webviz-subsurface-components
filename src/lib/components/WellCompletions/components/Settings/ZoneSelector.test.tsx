import value from "*.glsl";
import "@testing-library/jest-dom/extend-expect";
import { fireEvent, render, screen, within } from "@testing-library/react";
import userEvent from '@testing-library/user-event'
import React from "react";
import { testStore, Wrapper } from "../../test/TestWrapper";
import ZoneSelector from "./ZoneSelector";


describe("test Zone Selector", () => {
    it("snapshot test", () => {
        const { container } = render(
            Wrapper({ children: <ZoneSelector /> })
        );
        expect(container.firstChild).toMatchSnapshot();
    });

    it("click to dispatch redux action", async () => {
        const {container} = render(<ZoneSelector />, {
            wrapper: Wrapper,
        });
        userEvent.click(screen.getByRole('button', { name: "Select Zones â€‹" }))
        await screen.getByRole('listbox')

        //incomplete test
        /*
        const list = screen.getByRole('listbox')

        const { getByRole } = within(list)
        const items = getByRole("option", {name: 'zone1' , hidden: true})

        userEvent.click(screen.getByText('zone1'))
        */

        /*
        const list = screen.getByRole("listbox", {
            name: /Select Zones/i,})
            
        userEvent.selectOptions(list, ['zone1', 'zone2'])
        const { getAllByRole } = within(list)
        const items = getAllByRole('listbo')
        userEvent.click(items[0])
        */

        /*
        expect(testStore.dispatch).toHaveBeenCalledTimes(1);
        expect(testStore.dispatch).toBeCalledWith({
            payload: ["zone1", "zone2"],
            type: "ui/updateFilteredZones",
        
        });
        */
    });
});

