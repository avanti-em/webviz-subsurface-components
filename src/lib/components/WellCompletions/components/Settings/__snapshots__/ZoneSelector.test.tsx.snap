// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Zone Selector snapshot test 1`] = `
<div
  class="MuiFormControl-root makeStyles-formControl-2"
>
  <label
    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
    data-shrink="false"
    id="mutiple-zone-label"
  >
    Select Zones
  </label>
  <div
    class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    style="width: 230px;"
  >
    <div
      aria-haspopup="listbox"
      aria-labelledby="mutiple-zone-label demo-mutiple-zone"
      class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
      id="demo-mutiple-zone"
      role="button"
      tabindex="0"
    >
      <span>
        â€‹
      </span>
    </div>
    <input
      aria-hidden="true"
      class="MuiSelect-nativeInput"
      tabindex="-1"
      value=""
    />
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSelect-icon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;
